generator/src/main/java/com/scottlogic/deg/generator/Guice/DecisionTreeWalkerProvider.java

```diff
@@ -3,34 +3,41 @@
 import com.google.inject.Inject;
 import com.google.inject.Provider;
 import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
 import com.scottlogic.deg.generator.generation.GenerationConfigSource;
-import com.scottlogic.deg.generator.walker.DecisionTreeWalker;
+import com.scottlogic.deg.generator.walker.*;
 
 public class DecisionTreeWalkerProvider implements Provider<DecisionTreeWalker> {
```

Comment:
Is this still a provider or is this now a factory?

Reply:
Its still a provider, as its used by Google Guice to determine the appropriate `DecisionTreeWalker` that should be injected into other types, e.g. the `DecisionTreeDataGenerator`.